integrations/upgrade/index.test.ts

```diff
@@ -1545,3 +1545,87 @@ test(
     `)
   },
 )
+
+test(
+  'that it attaches the correct layers to the imported files',
+  {
+    fs: {
+      'package.json': json`
+        {
+          "dependencies": {
+            "tailwindcss": "workspace:^",
+            "@tailwindcss/upgrade": "workspace:^"
+          }
+        }
+      `,
+      'tailwind.config.js': js`module.exports = {}`,
+      'src/index.css': css`
+        @import 'tailwindcss/utilities';
+
+        /* No layer expected */
+        @import './my-components.css';
+
+        /* No layer expected */
+        @import './my-utilities.css';
+
+        /* Expecting a layer */
+        @import './my-other.css';
+
+        @import 'tailwindcss/components';
```

Comment:
Is it important that `components` comes last here? If that's not a crucial part of the test it seems better to me to structure things closer to how they would be in a real project.